### PR TITLE
integrate new locale

### PR DESCRIPTION
**Pull request by DmytroHryshyn**
## What does this PR do?

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/app/layout.tsx [hunk 1]
```diff
@@ -40,32 +41,44 @@ export const metadata: Metadata = {
...
+) => {
   const { pathname, searchParams } = new URL(url);
 
   const isEmbed = pathname.endsWith("/embed") || (searchParams?.get("embedType") ?? null) !== null;
   const embedColorScheme = searchParams?.get("ui.color-scheme");
 
-  // @TODO locale will be implemented during i18n migration
-  return { isEmbed, embedColorScheme, locale: "en" };
+  // @ts-expect-error we cannot access ctx.req in app dir, however headers and cookies are only properties needed to extract the locale
+  const newLocale = await getLocale({ headers, cookies });
```
Comment (grzpab) [inline apps/web/app/layout.tsx:55]: can we somewhat change the type of `getLocale` to make it work?

File: apps/web/app/layout.tsx [hunk 2]
```diff
@@ -40,32 +41,44 @@ export const metadata: Metadata = {
...
+
+  const intlLocale = new Intl.Locale(newLocale);
```
Comment (grzpab) [inline apps/web/app/layout.tsx:57]: this could throw for e.g. `*`, can you try-catch it?

File: apps/web/app/layout.tsx [hunk 3]
```diff
@@ -40,32 +41,44 @@ export const metadata: Metadata = {
...
+  // @ts-expect-error INFO: Typescript does not know about the Intl.Locale textInfo attribute
+  const direction = intlLocale.textInfo?.direction;
+
+  if (!direction) {
```
Comment (grzpab) [inline apps/web/app/layout.tsx:61]: if this cannot be figured out, it should be `ltr`

File: apps/web/lib/app-providers-app-dir.tsx [hunk 4]
```diff
@@ -59,7 +59,26 @@ const CustomI18nextProvider = (props: { children: React.ReactElement; i18n?: SSR
    **/
   // @TODO
 
-  const clientViewerI18n = useClientViewerI18n(["en"]);
+  const session = useSession();
+  const locale = session?.data?.user.locale ?? "en";
```
Comment (grzpab) [inline apps/web/lib/app-providers-app-dir.tsx:63]: what if there is no session? I think `window.document.documentElement.lang` should be already set to some value?

File: apps/web/app/layout.tsx (hunk 1 above)
Comment (DmytroHryshyn) [reply apps/web/app/layout.tsx:55]: i think no, because getLocale uses getToken that accepts NextRequest type.

Comment (grzpab) [reply apps/web/app/layout.tsx:55]: lemme check the code of the `getToken` now, I wonder if the shape of the object is correct

Comment (grzpab) [reply apps/web/app/layout.tsx:55]: alright, I checked, it only uses `cookies` and `headers` out of the `req` object, so all good. we can write that we manually checked it.

File: apps/web/lib/app-providers-app-dir.tsx [hunk 5]
```diff
@@ -59,7 +59,27 @@ const CustomI18nextProvider = (props: { children: React.ReactElement; i18n?: SSR
    **/
   // @TODO
 
-  const clientViewerI18n = useClientViewerI18n(["en"]);
+  const session = useSession();
+  const locale =
+    session?.data?.user.locale ?? typeof window !== "undefined" ? window.document.documentElement.lang : "en";
```
Comment (grzpab) [inline apps/web/lib/app-providers-app-dir.tsx:64]: I do wonder, can we calculate `dir` server side and pass it as props, similar to `newLocale`

Comment (DmytroHryshyn) [reply apps/web/lib/app-providers-app-dir.tsx:64]: we already calculate it on server side.  this hook is needed only for the case when we switch lang in settings on the client side

**Review by grzpab — APPROVED** [review]

**Pull request merged**